File: src/main/java/com/interview/SuperHeroMissionSG/model/SuperHero.java
package com.interview.SuperHeroMissionSG.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import org.hibernate.annotations.ResultCheckStyle;
import org.hibernate.annotations.SQLDelete;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "super_hero")
@Data
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@SQLDelete(sql = "DELETE FROM super_hero WHERE id=?", check = ResultCheckStyle.COUNT)
public class SuperHero {
    @Id
    @GeneratedValue(strategy=GenerationType.AUTO)
    private Long id;
    @Column(name = "superHeroName")
    private String superHeroName;
    @Column(name = "firstName")
    private String firstName;
    @Column(name = "lastName")
    private String lastName;
    @Column(name = "missionName")
    private String missionName;
}

File: webapp/src/app/superhero/superhero.ts
import {OnInit} from "@angular/core"
export class SuperHero implements OnInit{
  missionName: string;
  superHeroName: string;
  firstName: string;
  lastName: string;
  _missionName: string;
  _superHeroName: string;
  _firstName: string;
  _lastName: string;
  ngOnInit(): void {
      this.setFirstName('');
      this.setLastName('');
      this.setMissionName('');
      this.setSuperHeroName('');
  }
  public  _setMissionName(name: string){
    this._missionName = name;
  }
  public  _setSuperHeroName(name: string){
    this._superHeroName = name;
  }
  public  setMissionName(name: string){
    this.missionName = name;
  }
  public  setSuperHeroName(name: string){
    this.superHeroName = name;
  }
  public  _setFirstName(name: string){
    this._firstName = name;
  }
  public  _setLastName(name: string){
    this._lastName = name;
  }
  public  setFirstName(name: string){
    this.firstName = name;
  }
  public  setLastName(name: string){
    this.lastName = name;
  }

}
